### PR TITLE
docs: contributing

### LINKED ISSUE #5: Git repo documentation needed
probably this should be split into multible issues:

at the moment there is only a very brief introduktion how to install the project, to open source it and also for onboarding of developers there will be needed a lot of documentation:

1. repo structure: what documents and files go where
2. how to report bugs and features: idealy a bug template, what labels to use and when to use them...
3. comitment strategie: should you fork and make a pull/merge request or clone and push directly, who can do what (rols)
4. code style: what styles should be used in the code, are there tools that auto format or enforce styles?
5. how to commit: messages, when to branch, what branch is for what...
6. code of conduct
7. usfull tools and toolsets
8. for newcommers links to documentation and learning material for the different used technologies

### PR DESCRIPTION
**Pull request by razbakov**
closes #5

### CODE REVIEW

**Comment by bela127** [comment]
Nice work!
I like it ☺️

File: README.md [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
-- Sign in (Google or magic link)
-- Create profile
-- Feed (list of posts)
-- Community (list of profiles)
+## Built with
 
-Admin:
+- [Nuxt.js](https://nuxtjs.org) – Vuejs framework, which is used as Static Site Generator.
+- [Tailwind CSS](https://tailwindcss.com/) – utility-first CSS framework.
+- [Nuxt Composition Api](https://composition-api.nuxtjs.org/) – provides a way to use the Vue 3 Composition API in with Nuxt-specific features.
```
Comment (bela127) [inline README.md:9]: Little typo: in with

File: README.md [hunk 2]
```diff
@@ -1,36 +1,34 @@
...
 
-- [Vue Composition Api](https://composition-api.vuejs.org/)
-- [Firebase](https://firebase.google.com/)
-- [Nuxt.js](https://nuxtjs.org)
-- [Tailwind CSS](https://tailwindcss.com/)
+- [Yarn](https://classic.yarnpkg.com/en/docs/install) – javascript package manager
+- [VSCode](https://code.visualstudio.com/) – code editor
 
-## Setup
+### Setup
```
Comment (bela127) [inline README.md:22]:
```suggestion
### Setup
```
Is this the minimal setup that is sufficient to test and run and develop?
If it is call it minimal setup, and say what you can do and what you can't do with this setup

File: README.md [hunk 3]
```diff
@@ -1,36 +1,34 @@
...
+2. Clone your forked repository with `git clone https://github.com/<your-username>/wedance.git`
+3. Install dependencies with `yarn install`
+4. Run `yarn dev` to serve site with hot reload at https://localhost:3000/
 
-# serve with hot reload at localhost:3000
-yarn dev
-```
+## Site structure
+
+This site is a [Nuxt.js](https://nuxtjs.org) application. See [Directory Structure](https://nuxtjs.org/guide/directory-structure) and official guide for more information. Also most of folders in this repository has a `README.md` file.
```
Comment (bela127) [inline README.md:31]: Mention the "functions" folder containing serverles functions ?

File: README.md [hunk 4]
```diff
@@ -1,36 +1,34 @@
...
 
 ## Deploy
```
Comment (bela127) [inline README.md:33]: Is there a minimal deployment (for demonstration purpose) vs a full deployment?

File: README.md [hunk 5]
```diff
@@ -58,3 +56,74 @@ NUXT_ENV_GOOGLE_FIREBASE_API_KEY=
...
+
+Building features requires a lot of communication, but we'd love to have your help with this too! Features tend to be subjective and might spur some debate. Be sure to create an issue for new features before getting started! If your feature involves design changes, including design mockups can be very helpful. As always, when in doubt, ask!
+
+### How to contribute
+
+We'd love to see your pull requests, even if it's just to fix a typo!
+
+However, any significant improvement should be associated to an existing feature request or bug report.
+
+#### Getting started
```
Comment (bela127) [inline README.md:108]: Is this maybe the minimum/demo deployment?

File: README.md (hunk 2 above)
Comment (razbakov) [reply README.md:22]: It's the only setup. Same as described in Nuxt.js documentation 😉

**Review by bela127 — COMMENTED** [review]
I like it, I added some notes, tell me what you think, but I think even without them it makes a nice project introduction!

File: README.md (hunk 2 above)
Comment (bela127) [reply README.md:22]: I mean when you do this, can you already test all functionality of the side?
What can you already test?
What not?
When do you need a deployment?

File: README.md (hunk 4 above)
Comment (razbakov) [reply README.md:33]: There is no minimal deployment. Environment variables (API keys) are always required.

File: README.md (hunk 5 above)
Comment (razbakov) [reply README.md:108]: It's a step by step guide how to create a pull request.

Comment (bela127) [reply README.md:108]: Sorry it should have point to the text below 'getting started'

File: README.md (hunk 2 above)
Comment (razbakov) [reply README.md:28]: I added section `Services` which tells which services require more configuration.
Deployment is needed to provide a public link to your instance.

File: README.md [hunk 6]
```diff
@@ -1,60 +1,196 @@
...
-```
+1. Fork this repository, e.g. https://github.com/razbakov/wedance/fork
+2. Clone your forked repository with `git clone https://github.com/<your-username>/wedance.git`
+3. Install dependencies with `yarn install`
+4. Set up keys for your local instance of WeDance, you'll need to create an `.env` file. Take a look at `.env.example`. This file lists all the `ENV` variables we use and provides a fake default for any missing keys.
+5. Run `yarn dev` to serve site with hot reload at https://localhost:3000/
 
-## Deploy
+To activate all services and features see section `Services` below.
+(This is not needed for basic testing, but only for a full-scale deployment)
```
Comment (bela127) [inline README.md:44]: Is this correct?

File: README.md [hunk 7]
```diff
@@ -1,60 +1,196 @@
...
-- Accounts
-- Emails
+## Installation Guide
+In the following e detailed explanation is given, how to setup a running development version of the hole Projekt, with database and webhost.
+### Prerequisites
+What you need before you can begin.
 
-## Built with
+- [Yarn](https://classic.yarnpkg.com/en/docs/install) – javascript package manager
+- [Firebase](https://firebase.google.com/) - a firebase account
```
Comment (bela127) [inline README.md:22]: I added this as a prerequisite, is it ok?

File: README.md [hunk 8]
```diff
@@ -1,60 +1,196 @@
...
+- Publish directory: `dist`.
+- Click `Advanced build settings` and `New variable`.
+- Add all keys and values from `.env` file (`URL` and [some other variables](https://docs.netlify.com/configure-builds/environment-variables/) are set automatically)
+- Click `Deploy site`.
+
+To activate all services and features see section `Services` below.
+
+## Services
+If you want to run all services that are used for a fully functional projekt your right here.
+You don't need to do this if your good with testing without the services.
```
Comment (bela127) [inline README.md:87]: Is this correct?

**Review by bela127 — COMMENTED** [review]
I wrote some explanational texts, can you look over everything if everything is correct?

File: README.md (hunk 7 above)
Comment (razbakov) [reply README.md:22]: This section is `Built with` ;)

File: README.md (hunk 6 above)
Comment (razbakov) [reply README.md:44]: It depends what you are going to test, services are grouped by features.

File: README.md [hunk 9]
```diff
@@ -1,60 +1,196 @@
...
+- [Sentry](https://sentry.io/) – monitoring platform to log exceptions and errors.
+- [Netlify](https://netlify.com/) – CDN, CI and Hosting provider, that deploys site on each commit.
 
-- Reports
-- Tags
-- Profiles
-- Accounts
-- Emails
+## Installation Guide
+In the following e detailed explanation is given, how to setup a running development version of the hole Projekt, with database and webhost.
```
Comment (razbakov) [inline README.md:17]:
```suggestion
How to setup project locally with database.
```

File: README.md (hunk 7 above)
Comment (razbakov) [inline README.md:22]:
```suggestion
- Google Account
```

File: README.md [hunk 10]
```diff
@@ -1,60 +1,196 @@
...
+### Prerequisites
+What you need before you can begin.
 
-## Built with
+- [Yarn](https://classic.yarnpkg.com/en/docs/install) – javascript package manager
+- [Firebase](https://firebase.google.com/) - a firebase account
+- [Netlify](https://www.netlify.com/) - a Netlify account
```
Comment (razbakov) [inline README.md:23]: You don't need a Netlify account to setup project locally.

Comment (bela127) [reply README.md:23]: Do you need it for testing ?
Or can you run and test locally ?

**Comment by razbakov** [comment]
@bela127 could you please either use Suggestions via github UI or create a new branch? It's a bit difficult to review when our commits are mixed.

File: README.md (hunk 7 above)
Comment (bela127) [reply README.md:22]: Do you not need a firebase account ?
I thought you need one for setup?

File: README.md (hunk 10 above)
Comment (razbakov) [reply README.md]: It's hosting ;)

File: README.md (hunk 7 above)
Comment (razbakov) [reply README.md]: Firebase is using Google Accounts.

File: README.md (hunk 10 above)
Comment (bela127) [reply README.md]: Yes I know 😃
I mean, do you need to host it to fully test it (because some parts probably won't run locally)?

**Review by bela127 — APPROVED** [review]

**Pull request merged**